op-e2e/actions/kromampt_fork_test.go

```diff
@@ -0,0 +1,186 @@
+	dp.DeployConfig.L2GenesisRegolithTimeOffset = &genesisBlock
+	dp.DeployConfig.L2GenesisCanyonTimeOffset = &genesisBlock
+	dp.DeployConfig.L2GenesisDeltaTimeOffset = &genesisBlock
+	dp.DeployConfig.L2GenesisEcotoneTimeOffset = &genesisBlock
+	dp.DeployConfig.L2GenesisKromaMPTTimeOffset = &kromaMPTOffset
+	require.NoError(t, dp.DeployConfig.Check(), "must have valid config")
+
+	sd := e2eutils.Setup(t, dp, defaultAlloc)
+	log := testlog.Logger(t, log.LevelDebug)
+	_, _, _, sequencer, engine, verifier, _, _ := setupReorgTestActors(t, dp, sd, log)
+	ethCl := engine.EthClient()
+
+	// start op-nodes
+	sequencer.ActL2PipelineFull(t)
+	verifier.ActL2PipelineFull(t)
+
+	// Get gas price from oracle
+	gasPriceOracle, err := bindings.NewGasPriceOracleCaller(predeploys.GasPriceOracleAddr, ethCl)
+	require.NoError(t, err)
+
+	// Get fee vault recipients
+	protocolFeeVault, err := bindings.NewProtocolVaultCaller(predeploys.ProtocolVaultAddr, ethCl)
+	require.NoError(t, err)
+	protocolFeeRecipient, err := protocolFeeVault.RECIPIENT(nil)
+	require.NoError(t, err)
+
+	l1FeeVault, err := bindings.NewL1FeeVaultCaller(predeploys.L1FeeVaultAddr, ethCl)
+	require.NoError(t, err)
+	l1FeeRecipient, err := l1FeeVault.RECIPIENT(nil)
+	require.NoError(t, err)
+
+	// Get current implementations addresses (by slot) for L1Block + GasPriceOracle
+	initialL1BlockAddress, err := ethCl.StorageAt(context.Background(), predeploys.L1BlockAddr, genesis.ImplementationSlot, nil)
+	require.NoError(t, err)
+	initialGasPriceOracleAddress, err := ethCl.StorageAt(context.Background(), predeploys.GasPriceOracleAddr, genesis.ImplementationSlot, nil)
+	require.NoError(t, err)
+
+	// Build to the pre-KromaMPT block
+	sequencer.ActBuildL2ToPreKromaMPT(t)
+
+	// get latest block
+	latestBlock, err := ethCl.BlockByNumber(context.Background(), nil)
+	require.NoError(t, err)
+	require.Equal(t, sequencer.L2Unsafe().Number, latestBlock.Number().Uint64())
+
+	transactions := latestBlock.Transactions()
+	// L1Block: 1 set-L1-info + 5 deploys + 5 upgradeTo + 1 enable KromaMPT on GPO
+	// See [derive.KromaMPTNetworkUpgradeTransactions]
+	require.Equal(t, 12, len(transactions))
+
+	// Use KromaDepositTx instead of DepositTx
+	for i, otx := range transactions {
+		if otx.Type() == types.DepositTxType {
+			transactions[i], err = otx.ToKromaDepositTx()
+			require.NoError(t, err)
+		}
+	}
+
+	_, err = derive.L1BlockInfoFromBytes(sd.RollupCfg, latestBlock.Time(), transactions[0].Data())
+	require.NoError(t, err)
+	require.Equal(t, derive.L1InfoEcotoneLen, len(transactions[0].Data()))
+
+	// All transactions are successful
+	for i := 1; i < 12; i++ {
+		txn := transactions[i]
+		receipt, err := ethCl.TransactionReceipt(context.Background(), txn.Hash())
+		require.NoError(t, err)
+		if receipt.Status != types.ReceiptStatusSuccessful {
+			log.Error("transaction failed", "hash", txn.Hash().Hex(), "status", receipt.Status)
+			log.Info("transaction recceipt", "receipt", receipt)
```

Comment:
The `types.Receipt` does not have a `String()`, so it will be difficult to read.

Reply:
Oh these were logs for testing, I'll remove relative lines. Thanks!